File: lib/limelight/commands/unpack_command.rb
#- Copyright © 2008-2010 8th Light, Inc. All Rights Reserved.
#- Limelight and all included source files are distributed under terms of the GNU LGPL.

require 'limelight/commands/command'

module Limelight
  module Commands

    # See the following usage summary.
    #
    #  Usage: limelight unpack <production_path>
    #      Unpack a limelight production into the tmp directory.
    #      options:
    #      -h, --help                       Prints this usage summary.
    #
    class UnpackCommand < Command

      install_as "unpack"

      def self.description #:nodoc:
        return "Unpack a limelight production into the tmp directory."
      end

      protected ###########################################

      def parameter_description #:nodoc:
        return "<production_path>"
      end

      def parse_remainder(args) #:nodoc:
        @production_path = args.shift
        raise "Missing production path" if @production_path.nil?
      end

      def process #:nodoc:
        Main.initialize_temp_directory
        packer = Limelight::Util::Packer.new
        location = packer.unpack(@production_path)
        puts "Production was unpacked to: #{location}"
      end

    end
  end
end

File: src/limelight/commands/UnpackCommand.java
package limelight.commands;

import limelight.Boot;
import limelight.Context;
import limelight.io.FileUtil;
import limelight.io.Packer;

import java.util.Map;

public class UnpackCommand extends Command
{
  private static Arguments arguments;
  private Packer packer;

  public static Arguments arguments()
  {
    if(arguments == null)
    {
      arguments = new Arguments();
      arguments.addParameter("production", "Path to production file (.llp)");
      arguments.addValueOption("d", "destination", "dir", "Directory where the package will be unpacked.  Defaults to the current working directory (.).");
    }
    return arguments;
  }

  private Packer getPacker()
  {
    if(packer == null)
      packer = new Packer();
    return packer;
  }

  public void setPacker(Packer packer)
  {
    this.packer = packer;
  }

  @Override
  public void doExecute(Map<String, String> args)
  {
    String llpFile = args.get("production");
    String destination = args.get("destination");
    if(destination != null)
      getPacker().unpack(llpFile, destination);
    else
      getPacker().unpack(llpFile, Context.fs().workingDir());
  }

  @Override
  public String description()
  {
    return "Unpack the contents of a .llp production package.";
  }

  @Override
  public String name()
  {
    return "unpack";
  }

  @Override
  public Arguments getArguments()
  {
    return arguments();
  }
}
